Firmware/Marlin_main.cpp

```diff
@@ -919,16 +919,16 @@ void update_sec_lang_from_external_flash()
 {
 	if ((boot_app_magic == BOOT_APP_MAGIC) && (boot_app_flags & BOOT_APP_FLG_USER0))
 	{
-		uint8_t lang = boot_reserved >> 4;
-		uint8_t state = boot_reserved & 0xf;
+		uint8_t lang = boot_reserved >> 3;
+		uint8_t state = boot_reserved & 0x07;
 		lang_table_header_t header;
 		uint32_t src_addr;
 		if (lang_get_header(lang, &header, &src_addr))
 		{
 			lcd_puts_at_P(1,3,PSTR("Language update."));
 			for (uint8_t i = 0; i < state; i++) fputc('.', lcdout);
 			_delay(100);
-			boot_reserved = (state + 1) | (lang << 4);
+			boot_reserved = (boot_reserved & 0xF8) | ((state + 1) & 0x07);
```

Comment:
```
state=boot_reserver & 0x7; // up to value of 7, right?
... then nothing changes state
boot_reserved=(boot_reserved & 0xF8) | ((state + 1) & 0x07);
```
in the last line: state is still up to 7
state+1 is 8
8 & 7 is zero (i.e. nothing is added to `boot_reserved & 0xf8`)

is that correct?

Reply:
It won't matter. State 8 won't ever run. Right now we only use up to 4 states anyway. It will indeed save state 0 instead of 8, but it will also finish the whole language on state 7 at most

Reply:
ok